src/cascadia/TerminalCore/TerminalApi.cpp

```diff
@@ -193,7 +193,10 @@ void Terminal::PlayMidiNote(const int noteNumber, const int velocity, const std:
 
 void Terminal::UseAlternateScreenBuffer(const TextAttribute& attrs)
 {
-    _assertLocked();
+    if (_inAltBuffer())
+    {
+        return;
+    }

```

Comment:
This still isn't right. You can enable the alt buffer multiple times, but it just shouldn't be copying the cursor position from the main buffer every time that happens. It's more complicated than it should be because our buffer architecture doesn't align with xterm buffers (we store the cursor details as part of the buffer, but it's essentially meant to be a shared resource).

But ignore that for the moment, because I think there is a deeper issue here. When I try to enable the alt buffer multiple times, the second request doesn't seem to be passed through to WT at all. This is my test case:
```
printf "MAIN"; sleep 1; printf "\e[?1049hALT 1"; sleep 1; printf "\e[?1049hALT 2"; sleep 1; printf "\e[?1049l back to MAIN\n"
```
To see how it's meant to look, just run that in OpenConsole or the main branch of WT. In this branch, it's not showing ALT 2, and it also doesn't switch back to MAIN at the end.

I'm wondering if it's got something to do with the alt buffer changing the active `SCREEN_INFORMATION` instance, and that's breaking the `GetVtIo` implementation.

Reply:
> This still isn't right. You can enable the alt buffer multiple times, but it just shouldn't be copying the cursor position from the main buffer every time that happens.

Oh, I see. I'll mark this up as a follow-up fix then. The latest commit should fix this bug.

By the way, thank you so much for testing this PR and explaining these things to me!